### PR TITLE
Feat/app minimize

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

## Description

Capacitor (finally) released functionality to allow the app to be minimised (briefly could only be fully closed). As a result when pressing back button on the home screen running on an android device nothing happens.

This PR changes behaviour so that if back button is pressed on the home screen the app minimises (as is common for most other apps).

## Review Notes
As this can only be tested on a physical device recommend using generated link to test via appetize.io

## Dev Notes
Various core plugins have been updated to include the new feature. These are all minor version updates and so not expected to have any breaking changes.

## Git Issues

Closes #

## Screenshots/Videos

https://user-images.githubusercontent.com/10515065/150627860-fe02ca1a-ad7b-4d86-b893-d8ccb80b537e.mp4

### CODE REVIEW

File: packages/data-models/constants.ts [hunk 1]
```diff
@@ -81,6 +81,9 @@ export const APP_HEADER_DEFAULTS = {
   // default show back button on all screens except home screen
   should_show_back_button: (location: Location) =>
     location.pathname !== APP_ROUTE_DEFAULTS.home_path,
+  // on device minimize app when back button pressed from home screen
+  should_minimize_app_on_back: (location: Location) =>
+    location.pathname == APP_ROUTE_DEFAULTS.home_path,
 };
```
Comment (chrismclarke) [inline packages/data-models/constants.ts:87]: authoring option that handles which route to consider home page and minimise on back button push

File: package.json [hunk 2]
```diff
@@ -37,22 +37,22 @@
...
-    "@capacitor/push-notifications": "^1.0.4",
-    "@capacitor/splash-screen": "^1.1.2",
-    "@ionic-native/core": "^5.33.1",
-    "@ionic-native/device": "^5.33.1",
-    "@ionic-native/http": "^5.33.1",
-    "@ionic-native/media": "^5.33.1",
-    "@ionic-native/status-bar": "^5.33.1",
-    "@ionic/angular": "^6.0.1",
-    "@ionic/pwa-elements": "^3.0.2",
+    "@capacitor/android": "^3.4.0",
```
Comment (chrismclarke) [inline package.json:40]: All package updates are minor version bumps only, so not expected to introduce any breaking changes

File: src/app/shared/components/plh-main-header.ts [hunk 3]
```diff
@@ -43,26 +48,43 @@ export class PLHMainHeaderComponent implements OnInit, OnDestroy {
       }
     });
     this.handleRouteChange();
+    // subscribe to and handle hardware back button press
+    if (Capacitor.getPlatform() === "android") {
+      this.hardwareBackButton$ = await App.addListener("backButton", () => {
+        this.handleHardwareBackPress();
+      });
```
Comment (chrismclarke) [inline src/app/shared/components/plh-main-header.ts:55]: listener to add handling to hardware back button press

File: android/app/capacitor.build.gradle [hunk 4]
```diff
@@ -10,6 +10,7 @@ android {
 apply from: "../capacitor-cordova-android-plugins/cordova.variables.gradle"
 dependencies {
     implementation project(':capacitor-app')
+    implementation project(':capacitor-clipboard')
```
Comment (chrismclarke) [inline android/app/capacitor.build.gradle:13]: when we previously added clipboard copy feature some of the linked dependencies were not tracked in main repo. This isn't really an issue as they are re-populated automatically during build on the server, but added now for tidyness

**Comment by EsmeeIDEMS** [comment]
I get this message about half of the time when I refresh the screen in Appetize. Is that expected? Should I create my own account?

image

**Review by EsmeeIDEMS — CHANGES_REQUESTED** [review]
When testing on Appetize, it seems that only the first action on the Language page works: when "next" is clicked immediately, the page closes correctly, but when first a language is selected, neither the radio group nor the "next" button responds any longer. 

(N.B. on the Terms and Conditions page multiple actions do seem to work, after launching pop-ups the "next" button still works.)

App gets stuck when interacting with language radio group:

https://user-images.githubusercontent.com/74557272/150756194-d4111ebc-f380-48ab-8903-b5bbadd5609c.mp4

App doesn't get stuck when not interacting with language radio group:

https://user-images.githubusercontent.com/74557272/150756626-45228ea6-3c73-456f-9920-6459a86f8139.mp4

**Comment by chrismclarke** [comment]
> When testing on Appetize, it seems that only the first action on the Language page works: when "next" is clicked immediately, the page closes correctly, but when first a language is selected, neither the radio group nor the "next" button responds any longer.
> 

I get the same issue locally so it's not specific to appetize. Will convert to a separate issue and suggest merging here unless there are any other issues noted.

**Review by volloholic — APPROVED** [review]

**Review by EsmeeIDEMS — APPROVED** [review]

**Pull request merged**